fxrecorder/src/bin/main.rs

```diff
@@ -29,15 +33,39 @@ fn main() {
 }
 
 async fn fxrecorder(log: Logger, _options: Options, config: Config) -> Result<(), Box<dyn Error>> {
-    use libfxrecorder::proto::RecorderProto;
-    use tokio::net::TcpStream;
+    {
+        let stream = TcpStream::connect(&config.host).await?;
+        info!(log, "Connected"; "peer" => config.host);
 
-    let stream = TcpStream::connect(&config.host).await?;
-    info!(log, "Connected"; "peer" => config.host);
+        let mut proto = RecorderProto::new(log.clone(), stream);
 
-    let mut proto = RecorderProto::new(log, stream);
+        proto.handshake(true).await?;
+    }
+
+    {
+        let reconnect = || {
+            info!(log, "Attempting re-connection to runner...");
+            TcpStream::connect(&config.host)
```

Comment:
What is the default timeout for `TcpStream::connect`? I can't find any mention of a timeout in the docs - does it just block indefinitely (making the exponential backoff stuff redundant?)

Reply:
I believe it is OS dependent. When testing locally I definitely see multiple "Attempting re-connection" messages.